catalyst/experiments/hydra_config.py

```diff
@@ -0,0 +1,552 @@
+        # instantiate optimizer
+        optimizer: Optimizer = hydra.utils.instantiate(
+            params, params=model_params
+        )
+        # load from previous stage
+        if load_from_previous_stage and self.stages.index(stage) != 0:
+            checkpoint_path = f"{self.logdir}/checkpoints/best_full.pth"
+            checkpoint = load_checkpoint(checkpoint_path)
+            dict2load = optimizer
+            if name is not None:
+                dict2load = {name: optimizer}
+            unpack_checkpoint(checkpoint, optimizer=dict2load)
+            # move optimizer to device
+            device = get_device()
+            for param in model_params:
+                param = param["params"][0]
+                optimizer_state = optimizer.state[param]
+                for state_key, state_value in optimizer_state.items():
+                    optimizer_state[state_key] = any2device(
+                        state_value, device
+                    )
+            # update optimizer params
+            for key, value in params.items():
+                for optimizer_param_group in optimizer.param_groups:
+                    optimizer_param_group[key] = value
+
+        return optimizer
+
+    def get_optimizer(
+        self, stage: str, model: Dict[str, Model]
+    ) -> Dict[str, Optimizer]:
+        """
+        Returns the optimizers for a given stage.
+
+        Args:
+            stage (str):  stage name
+            model (Dict[str, Model]): dict of models
+
+        Returns:
+            Dict[str, Optimizer]: optimizer for selected stage
+
+        """
+        if "optimizers" not in self._config.stages[stage]:
+            return {}
+        optimizers_params: DictConfig = self._config.stages[stage].optimizers
+        optimizer = {
+            name: self._get_optimizer(stage, model, name, optimizer_params)
+            for name, optimizer_params in optimizers_params.items()
+        }
+        return optimizer
+
+    @staticmethod
+    def _get_scheduler(
+        optimizers: Dict[str, Optimizer], params: DictConfig
+    ) -> Scheduler:
+        assert (
+            "optimizer" in params
+        ), "scheduler config must contain 'optimizer' key"
+        optimizer_key: str = params.pop("optimizer")
+        scheduler: Scheduler = hydra.utils.instantiate(
+            params, optimizer=optimizers[optimizer_key]
+        )
+        return scheduler
+
+    def get_scheduler(
+        self, stage: str, optimizer: Dict[str, Optimizer]
+    ) -> Dict[str, Scheduler]:
+        """Returns the schedulers for a given stage."""
+        if "schedulers" not in self._config.stages[stage]:
+            return {}
+        schedulers_params: DictConfig = self._config.stages[stage].schedulers
+        scheduler: Dict[str, Scheduler] = {
+            key: self._get_scheduler(
+                optimizers=optimizer, params=scheduler_params
+            )
+            for key, scheduler_params in schedulers_params.items()
+        }
```

Comment:
is there any support for non-key-value scheduler?

Reply:
No. See previous answer.